Tools/peg_generator/pegen/c_generator.py

```diff
@@ -416,7 +417,8 @@ def out_of_memory_goto(self, expr: str, goto_target: str) -> None:
 
     def generate(self, filename: str) -> None:
         self.collect_rules()
-        self.print(f"// @generated by pegen from {filename}")
+        basename = os.path.basename(filename)
```

Comment:
Can we use pathlib instead of os.path?

```python
from pathlib import Path
# [...]
        basename = Path(filename).name
```

Reply:
``pathlib`` has a much more complex import tree and relies on more modules. ``os.path`` has the benefit that it works with a minimal interpreter at an early stage of the bootstrapping process.

Reply:
Yes, pathlib can be used, but I prefer os.path :-) For this specific change, I don't think that pathlib is much better.